### ISSUE TITLE
Units for traffic_light_penalty (and turn.duration)?

### ISSUE THREAD

**Issue by mattwigway** [Project-OSRM/osrm-backend#5989]
In [car.lua](https://github.com/Project-OSRM/osrm-backend/blob/master/profiles/car.lua#L29) the traffic light penalty is set to 2, and [in the documentation for profiles, it says traffic_light_penalty is in deciseconds](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/profiles.md), which would indicate that the default traffic light penalty is 0.2 seconds, which seems much too short. In [testbot.lua](https://github.com/Project-OSRM/osrm-backend/blob/master/profiles/testbot.lua#L18) there is a comment that says this parameter is in seconds. (In both profiles, the traffic_light_penalty is simply added to the turn duration if there is a traffic light). On [line 618 of edge_based_graph_factory.cpp](https://github.com/Project-OSRM/osrm-backend/blob/master/src/extractor/edge_based_graph_factory.cpp#L618) the turn duration is multiplied by 10 (rather than divided) which would suggest that the units of turn.duration are actually decaseconds (tens of seconds), rather than deciseconds (tenths of seconds).

**Comment by danpat** [comment]
Well, that's all a bit of a mess now isn't it....  PR making things consistent would be welcome :-)

For backward compatibility, we shouldn't change the behaviour and defaults, but we should certainly update the documentation.

**Comment by mattwigway** [comment]
Happy to make a PR, and agree with not changing defaults. If someone could put another pair of eyes in edge_based_graph_factory and confirm that the units do appear to be 10s of seconds, that would be much appreciated :)

**Comment by danpat** [comment]
@mattwigway Internally, all the duration values are in deciseconds.  The code in the `edge_based_graph_factory` is expecting `turn.duration` to be expressed in *seconds*.  We use deciseconds internally until we finally serialize the response, at points like [here](https://github.com/Project-OSRM/osrm-backend/blob/master/include/engine/guidance/assemble_leg.hpp#L215) and [here](https://github.com/Project-OSRM/osrm-backend/blob/master/include/engine/guidance/assemble_steps.hpp#L140) and finally [here, where we sum up the already divided leg.steps](https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/guidance/assemble_route.cpp#L18-L21).

However, the story goes a little deeper than that.  If you look in https://github.com/Project-OSRM/osrm-backend/blob/master/src/extractor/scripting_environment_lua.cpp#L1075-L1134 there's a bit of a story - different API versions inside the lua scripts have lead to different return values from the turn function.  The "latest" version of the Lua API expects the turn function to return values in seconds.

So the documentation looks like it matches Version 0 of the Lua API, where the return value *was* in deciseconds (and the scripting_environment_lua.cpp code divided the return result by 10).

TL;DR - docs need to be updated, the `turn_function` should return seconds, and the `traffic_light_penalty`, as it's simply added within the `turn_function`, is also in seconds.

**Comment by wangyoucao577** [comment]
Issue related https://github.com/Project-OSRM/osrm-backend/issues/5400
